magento/magento2
# 23411 remove misplaced critical log

### Description 
There is no longer a critical error when a logged in customer has no active quote - which is e.g. the case when an order was placed but no new quote was yet created.

@see https://github.com/magento/magento2/issues/23411#issuecomment-516611919

### Fixed Issues (if relevant)

1. magento/magento2#23411: Sorry, but always the same - after upgrading to new version something went wrong

### Manual testing scenarios
1. Register / log in as a customer in shop frontend
2. Add at least one product to the cart and proceed to checkout
3. Submit the order
    * *was:* A `report.CRITICAL: No such entity` is logged
    * *now:* no `CRITICAL` log appears
4. Navigate the catalog
    * *was:* More of these log entries appear
    * *now:* no `CRITICAL` log appears

### Questions or comments
Improves https://github.com/magento/magento2/commit/117d7103de2989168af45de550091e79f815d54e

### Contribution checklist
